luaui/Widgets/gui_gridmenu.lua

```diff
@@ -2221,6 +2255,7 @@ function widget:MousePress(x, y, button)
 			return true
 		end
 
+	-- i wish i had any idea what this code did
```

Comment:
Anything `preGamestartPlayer` is before the game starts (start box positions). Buildmenu is responsible for managing build queue in this circumstance since the engine does not store command queues (there is no unit to send commands to). When the game starts we send our internal queue to the unit that spawns.

Reply:
You can change it to a comment explaining what it does (now that you know) or remove it.